src/providers/TransactionsDispatcherContextProvider.tsx

```diff
@@ -153,54 +195,67 @@ const TransactionsDispatcherContextProvider = ({ children }: { children: ReactNo
           web3Provider,
           transactionsToSend[0],
           firstUnsentCrossChainAction.chainId,
-          providerAddress,
+          providerAddress
         )
       : await submitEtherspotTransactionsBatch(
           getSdkForChainId(firstUnsentCrossChainAction.chainId) as Sdk,
-          transactionsToSend,
+          transactionsToSend
         );
 
-    if (result?.errorMessage || (!result?.transactionHash?.length && !result?.batchHash?.length)) {
+    if (
+      result?.errorMessage ||
+      (!result?.transactionHash?.length && !result?.batchHash?.length)
+    ) {
       resetCrossChainActions();
       return;
     }
 
     const { transactionHash, batchHash } = result;
 
-    const updatedCrossChainActions = crossChainActions.map((crossChainAction) => {
-      if (crossChainAction.id !== firstUnsentCrossChainAction.id
-        || firstUnsentCrossChainAction.multiCallData?.id !== crossChainAction.multiCallData?.id) {
-        return crossChainAction;
-      }
+    const updatedCrossChainActions = crossChainActions.map(
+      (crossChainAction) => {
+        if (
+          crossChainAction.id !== firstUnsentCrossChainAction.id &&
+          (!crossChainAction.multiCallData?.id ||
+            firstUnsentCrossChainAction.multiCallData?.id !==
+              crossChainAction.multiCallData.id)
+        ) {
+          return crossChainAction;
+        }
```

Comment:
@poocart Basically changed the `||` back to `&&` but also added another check - the reason for this is that in a multicall, even after the tx had been signed, it still made us sign the subsequent steps, trigger the tx twice.





[3:40](https://pillarwallet.slack.com/archives/C04DD82G8G5/p1669995615580619)
This check makes sure that the current action to be executed is not part of a multicall that's already been executed

Reply:
Thanks